stdlib/public/core/Range.swift

```diff
@@ -14,7 +14,7 @@
 
 public struct RangeOfStrideable<
   Element : Strideable
-> : Equatable, Collection,
+> : Equatable, RandomAccessCollection,
```

Comment:
I don't see where we provide a bulk-advance() method on `RangeOfStrideable`.  If we only provide `next()`, then adding random access would cause us to violate the O(1) guarantee of `advance()`.  Could you add an implementation or a fixme?